### PR TITLE
Bump pack to v0.18.1 to pick up DOCKER_HOST fix

### LINKED ISSUE #1409: pack builder ignores DOCKER_HOST
**Describe the bug**

When using the pack builder the `DOCKER_HOST` environment variable doesn't seem to be respected.
This causes the build to fail when the Docker socket is on a different location as default `/var/run/docker`, for example when running in rootless mode.

Interestingly the variable is properly used when creating the initial build pack container 
https://github.com/hashicorp/waypoint/blob/b550e5cee58c3cf363dc5abe00c77b42a9a92a5a/builtin/pack/builder.go#L78

When I capture the analyzer container using `docker inspect` it shows the socket mount to the wrong path (abbreviated output);
```json
{
  "Id": "5baa0c8737efe0045d39d3f41b69dc5aed72204a383cf243fbdffb7c16d5746c",
  "Created": "2021-05-04T12:56:29.277985308Z",
  "Path": "/cnb/lifecycle/analyzer",
  "Args": [
    "-daemon",
    "-cache-dir",
    "/cache",
    "-layers",
    "/layers",
    "index.docker.io/library/go:latest"
  ],
  "HostConfig": {
    "Binds": [
      "/var/run/docker.sock:/var/run/docker.sock",
      "pack-cache-fec58b24b18f.build:/cache",
      "pack-layers-htmzbpxzfb:/layers",
      "pack-app-mekfvmheeo:/workspace"
    ]
  },
  "Mounts": [
    {
      "Type": "bind",
      "Source": "/var/run/docker.sock",
      "Destination": "/var/run/docker.sock",
      "Mode": "",
      "RW": true,
      "Propagation": "rprivate"
    }
  ],
  "Config": {
    "Env": [
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
      "CNB_USER_ID=1000",
      "CNB_GROUP_ID=1000",
      "STACK=heroku-18",
      "CNB_STACK_ID=heroku-18"
    ],
    "Cmd": [
      "/cnb/lifecycle/analyzer",
      "-daemon",
      "-cache-dir",
      "/cache",
      "-layers",
      "/layers",
      "index.docker.io/library/go:latest"
    ],
    "Image": "pack.local/builder/7474787872686a736b68:latest",
    "Volumes": null
  }
}
```

**Steps to Reproduce**
```bash
> echo $DOCKER_HOST
unix:///run/user/1000/docker.sock
```

```hcl
project = "testing"

app "go" {
    build {
        use "pack" {}
    }

    deploy {
        use "docker" {}
    }
}
```
```bash
> waypoint up -v
...
 │ 2021/05/04 14:48:31.390228 INFO:   ===> DETECTING
 │ [detector] ======== Output: heroku/ruby@0.0.1 ========
 │ [detector] no
 │ [detector] err:  heroku/ruby@0.0.1 (1)
 │ [detector] heroku/go       0.3.1
 │ [detector] heroku/procfile 0.6.2
 │ 2021/05/04 14:48:31.913333 INFO:   ===> ANALYZING
 │ [analyzer] ERROR: failed to initialize docker client: failed to connect to docke
 │ r socket: dial unix /var/run/docker.sock: connect: connection refused
! failed with status code: 1
2021-05-04T14:48:32.312+0200 [WARN]  waypoint: context cancelled, stopping interrupt listener loop
```

**Expected behavior**
Would it be possible for Waypoint to inform CNB about the different location?

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: `v0.3.1 (b550e5ce)`
* Waypoint Server Platform and Version: `v0.3.1`
* Waypoint Plugin: `pack` & `docker`

### PR DESCRIPTION
**Pull request by evanphx**
This also bumped docker's version, those some adjustments to docker APIs.

Fixes #1409

### CODE REVIEW

File: builtin/pack/builder.go [hunk 1]
```diff
@@ -133,39 +133,30 @@ func (b *Builder) Build(
...
 	step.Done()
 
+	var ignore []string
+
+	// We want a copy of the slice, not share it because concurrency within
+	// the runner, etc.
+	ignore = append(ignore, b.config.Ignore...)
+
+	if jobInfo.Local {
+		ignore = []string{"data.db", "data.db.lock"}
```
Comment (mitchellh) [inline builtin/pack/builder.go:143]: Is it even worth doing this? You can technically change these (unlikely anyone does) and I could see a REALLY weird bug/behavior driving someone angry years from now when this is getting ignored for some reason lol. These files (especiially local) are never big, so just not sure if its worth even doing.

File: go.mod [hunk 2]
```diff
@@ -121,10 +124,5 @@ require (
...
 
-replace (
-	// v0.3.11 panics for some reason on our tests
-	github.com/imdario/mergo => github.com/imdario/mergo v0.3.9
-
-	// https://github.com/ory/dockertest/issues/208
-	golang.org/x/sys => golang.org/x/sys v0.0.0-20200826173525-f9321e4c35a6
-)
+// v0.3.11 panics for some reason on our tests
+replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.9
```
Comment (mitchellh) [inline go.mod:128]: So the sys stuff is just fixed now?

File: builtin/pack/builder.go (hunk 1 above)
Comment (evanphx) [reply builtin/pack/builder.go:137]: Up to you! You did that originally.

File: go.mod (hunk 2 above)
Comment (evanphx) [reply go.mod:128]: Yup! everything compiles fine now, the offending package has been fixed.

Comment (mitchellh) [reply go.mod:128]: YES

**Review by mitchellh — APPROVED** [review]

**Review by briancain — APPROVED** [review]

**Pull request merged**